src/helpers/transactions.ts

```diff
@@ -1,46 +1,62 @@
-export const ERC721_BADGE_BATCH_MINTED_TYPE = "ERC721Badge Batch Mint";
-export const ERC721_BADGE_TRANSFER_TYPE = "ERC721Badge Transfer";
-export const ERC721_BADGE_BURN_TYPE = "ERC721Badge Burn";
-export const ERC721_BADGE_UPDATE_TYPE = "ERC721Badge Update";
-
-export const ERC721_MINTED_TYPE = "ERC721 Mint";
-export const ERC721_BATCH_MINTED_TYPE = "ERC721 Batch Mint";
-export const ERC721_TRANSFER_TYPE = "ERC721 Transfer";
-export const ERC721_BURN_TYPE = "ERC721 Burn";
-
-export const ERC20_MINT_TYPE = "ERC20 Mint";
-export const ERC20_BURN_TYPE = "ERC20 Burn";
-export const ERC20_TRANSFER_TYPE = "ERC20 Transfer";
-export const ERC20_UPDATE_TYPE = "ERC20 Update";
-
-export const TOKEN_CREATE_TYPE = "Token Create";
-export const TOKEN_MINT_TYPE = "Token Reward";
-export const TOKEN_TRANSFER_TYPE = "Token Transfer";
-
-export const BADGE_CREATE_TYPE = "Badge Create";
-export const BADGE_MINT_TYPE = "Badge Reward";
-export const BADGE_TRANSFER_TYPE = "Badge Transfer";
-
-export const APP_CREATE_TYPE = "App Create";
-
+import { TypedMap } from "@graphprotocol/graph-ts";
+
+// Event Types
+export const ERC721_LAZY_MINT_TYPE            = "ERC721Lazy Mint";
+export const ERC721_LAZY_BATCH_MINT_TYPE      = "ERC721Lazy Batch Mint";
+export const ERC721_LAZY_MINT_LAZY_TYPE       = "ERC721Lazy Lazy Mint";
+export const ERC721_LAZY_TRANSFER_TYPE        = "ERC721Lazy Transfer";
+export const ERC721_LAZY_BURN_TYPE            = "ERC721Lazy Burn";
+
+export const ERC721_BADGE_MINTED_TYPE         = "ERC721Badge Mint";
+export const ERC721_BADGE_BATCH_MINTED_TYPE   = "ERC721Badge Batch Mint";
+export const ERC721_BADGE_TRANSFER_TYPE       = "ERC721Badge Transfer";
+export const ERC721_BADGE_BURN_TYPE           = "ERC721Badge Burn";
+export const ERC721_BADGE_UPDATE_TYPE         = "ERC721Badge Update";
+
+export const ERC721_MINTED_TYPE               = "ERC721 Mint";
+export const ERC721_BATCH_MINTED_TYPE         = "ERC721 Batch Mint";
+export const ERC721_TRANSFER_TYPE             = "ERC721 Transfer";
+export const ERC721_BURN_TYPE                 = "ERC721 Burn";
+
+export const ERC20_MINT_TYPE                  = "ERC20 Mint";
+export const ERC20_BURN_TYPE                  = "ERC20 Burn";
+export const ERC20_TRANSFER_TYPE              = "ERC20 Transfer";
+export const ERC20_UPDATE_TYPE                = "ERC20 Update";
+
+export const TOKEN_CREATE_TYPE                = "Token Create";
+export const TOKEN_MINT_TYPE                  = "Token Reward";
+export const TOKEN_TRANSFER_TYPE              = "Token Transfer";
+
+export const BADGE_CREATE_TYPE                = "Badge Create";
+export const BADGE_MINT_TYPE                  = "Badge Reward";
+export const BADGE_TRANSFER_TYPE              = "Badge Transfer";
+
+export const APP_CREATE_TYPE                  = "App Create";
+
+// Transaction Types
+// Keys are function selectors from transaction.input
+const functionTypes = new TypedMap<string, string>();
+functionTypes.set("0xb92bafd6", "Batch Mint Badge");
```

Comment:
After playing with this I realise I left of create in the app factory contract.
Could you please add:

`functionTypes.set("0x663ff5a9", "Create App");`

Reply:
Added! :+1: